test/cli/utils_test.go

```diff
@@ -0,0 +1,118 @@
+package cli
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+
+	"github.com/acarl005/stripansi"
+)
+
+type traitAssertion func(tb testing.TB, stdout, stderr string, rc int)
+
+func assertInOutput(data string) traitAssertion {
+	return func(tb testing.TB, stdout, stderr string, _ int) {
+		if !strings.Contains(stripansi.Strip(stderr), data) && !strings.Contains(stripansi.Strip(stdout), data) {
+			tb.Errorf("data=%q was NOT found in any output, but should have been there", data)
+		}
+	}
+}
+
+func getGrypeCommand(t testing.TB, args ...string) *exec.Cmd {
+	var binaryLocation string
+	if os.Getenv("GRYPE_BINARY_LOCATION") != "" {
+		// GRYPE_BINARY_LOCATION is the absolute path to the snapshot binary
+		binaryLocation = os.Getenv("GRYPE_BINARY_LOCATION")
+	} else {
+		// note: there is a subtle - vs _ difference between these versions
+		switch runtime.GOOS {
+		case "darwin":
+			binaryLocation = path.Join(repoRoot(t), fmt.Sprintf("snapshot/grype-macos_darwin_%s/grype", runtime.GOARCH))
+		case "linux":
+			binaryLocation = path.Join(repoRoot(t), fmt.Sprintf("snapshot/grype_linux_%s/grype", runtime.GOARCH))
+		default:
+			t.Fatalf("unsupported OS: %s", runtime.GOOS)
+		}
+
+	}
+	cmd := exec.Command(binaryLocation, args...)
+	// note: we need to preserve env vars + add an additional var to suppress checking for app updates
+	cmd.Env = append(os.Environ(), "GRYPE_CHECK_FOR_APP_UPDATE=false")
+	return cmd
+}
+
+func runGrypeCommand(t testing.TB, env map[string]string, args ...string) (*exec.Cmd, string, string) {
+	cmd := getGrypeCommand(t, args...)
+	if env != nil {
+		cmd.Env = append(cmd.Env, envMapToSlice(env)...)
+	}
+	var stdout, stderr bytes.Buffer
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+
+	// ignore errors since this may be what the test expects
+	cmd.Run()
```

Comment:
Ah... I see, so in regards to my previous comment for normalizing, this could be handled by a wrapper to this function that when the command fails, the reporting is done automatically. Additionally, a separate wrapper function would be one that failure is expected and so it doesn't check `t.Failed()`. Naming is tricky here, but something like: `checkGrypeCommand` when failure shouldn't happen, and keep `runGrypeCommand` when it doesn't matter?

Reply:
a clarification: `t.Failed()` is with regard to the test status, not the status of the grype process or it's exit code.